card/mod/core/spec/set/all/event_conditions_spec.rb

```diff
@@ -51,6 +51,58 @@ def change_content
       end
     end
 
+    context "when changing sets" do
+      def update_type
+        Card::Auth.as_bot do
+          Card["Sample Pointer"].update! type: "Search"
+        end
+      end
+
+      it "does NOT run update events from sets that no longer apply after change" do
+        with_test_events do
+          test_event :validate, on: :update, set: Card::Set::Type::Pointer do
+            add_to_log "NO to run"
+          end
+          update_type
+          expect(@log).to be_empty
+        end
+      end
+
+      it "does run update events from old sets when `changing` value present" do
+        with_test_events do
+          test_event :validate,
+                     on: :update, set: Card::Set::Type::Pointer, changing: :type do
+            add_to_log "YES to run"
+          end
+          update_type
+          expect(@log).to contain_exactly("YES to run")
+        end
+      end
+
+      it "does run update events from sets that apply after change" do
+        with_test_events do
+          test_event :validate, on: :update, set: Card::Set::Type::SearchType do
+            add_to_log "YES to run"
+          end
+          update_type
+          expect(@log).to contain_exactly("YES to run")
+        end
+      end
+
+      # following does not yet work, because old card has both old sets and new sets
+      # when conditions are tested.
+      xit "does NOT run update events from new sets when `changing` value present" do
+        with_test_events do
+          test_event :validate,
+                     on: :update, set: Card::Set::Type::SearchType, changing: :type do
+            add_to_log "NO to run"
+          end
+          update_type
+          expect(@log).to be_empty
+        end
+      end
+    end
```

Comment:
How about name changes? 
I'm not sure if that's preferable but if you make the log messages more like the test description then you could put all events in one test.